### PR TITLE
feat(FileExistsChecker): distinguish between empty and non-readable files

### PR DESCRIPTION
**Pull request by chrisknedl**
### Additions:

- Non-readable files are now classified as non-readable instead of empty
- Refactored the file-exists checker to be more easily expandable in the future in case more categories are required still

### CODE REVIEW

File: src/main/java/de/uni_passau/fim/se2/pipeline_helper/checkers/file_exists/FileExistsChecker.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        Arrays.stream(FileStatus.values()).forEach(status -> filesByStatus.put(status, new ArrayList<>()));
+
+        for (final Path p : toCheck) {
+            if (!Files.exists(p)) {
+                filesByStatus.get(FileStatus.MISSING).add(p);
+                continue;
+            }
+
+            try {
+                if (Files.isRegularFile(p) && isFileEmpty(p)) {
```
Comment (b-fein) [inline src/main/java/de/uni_passau/fim/se2/pipeline_helper/checkers/file_exists/FileExistsChecker.java:42]: Creating a directory instead of a file the tool looks for produces a successful result:

```bash
mkdir Tests.txt
java -jar ./target/pipeline-helper-2.2.1-SNAPSHOT-shaded.jar file-exists Tests.txt
cat customFeedbacks/TEST-FileExistsChecker.json
```
shows
```json
{"name":"FileExistsChecker","successful":true}
```

**Review by b-fein — CHANGES_REQUESTED** [review]
Looks good. Thanks!

Found one relevant edge-case, though.

File: src/test/java/de/uni_passau/fim/se2/pipeline_helper/checkers/FileExistsCheckerTest.java [hunk 2]
```diff
@@ -65,12 +68,40 @@ void shouldRecogniseNonExistingFiles() throws Exception {
...
+
+        final CheckerResult result = checker.check();
+        assertThat(result.isSuccessful()).isFalse();
+        assertThat(result.getMessage()).isEqualTo("""
+            Non-readable files:
+            InvalidByteSequence.java""");
+    }
+
+    @Test
+    void shouldNotConsiderAnEmptyDirectoryAFile() throws Exception {
```
Comment (b-fein) [inline src/test/java/de/uni_passau/fim/se2/pipeline_helper/checkers/FileExistsCheckerTest.java:86]:
```suggestion
    void shouldNotConsiderADirectory() throws Exception {
```
The new `Tests.txt` directory is not empty. (I think you could have used any of the pre-existing directories for the test instead.)

Comment (chrisknedl) [reply src/test/java/de/uni_passau/fim/se2/pipeline_helper/checkers/FileExistsCheckerTest.java:86]: You're right 🤦‍♂️ Will do that then

**Review by b-fein — APPROVED** [review]
